WebDriverAgentLib/Utilities/XCUIApplicationProcessDelay.m

```diff
@@ -53,11 +58,17 @@ + (void)load {
   orig_set_event_loop_has_idled = (void(*)(id, SEL, BOOL)) method_getImplementation(original);
   Method replace = class_getClassMethod([XCUIApplicationProcessDelay class], @selector(setEventLoopHasIdled:));
   method_setImplementation(original, method_getImplementation(replace));
+  isEnabled = YES;
 }
 
 + (void)setEventLoopHasIdled:(BOOL)idled {
-  [FBLogger verboseLog:[NSString stringWithFormat:@"Delaying -[XCUIApplicationProcess setEventLoopHasIdled:] by %.2f seconds", delay]];
-  [NSThread sleepForTimeInterval:delay];
+  [lock lock];
```

Comment:
what is the purpose of this lock?

Reply:
cannot we just use `@syncronize` instead?

Reply:
`eventloopIdleDelay` is written and read from different threads.
I wanted to go with `@synchronized(self)`, but since we don't know whats going on in the original implementation of the class I was using a separate one